### PR TITLE
Adding DALI dataset

### PR DESCRIPTION
**Pull request by magdalenafuentes**

### CODE REVIEW

File: scripts/make_dali_index.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def make_dali_index(data_path):
+    metadata_path = os.path.join(data_path, 'dali_metadata.json')
+    with open(metadata_path, 'r') as fhandle:
+        dali_metadata = json.load(fhandle)
+    dali_index = {}
+    for trackid in dali_metadata.keys():
+        dali_index[trackid] = {}
+        dali_index[trackid]['audio'] = ['DALI/audio/'+trackid+'.mp3']
```
Comment (rabitt) [inline scripts/make_dali_index.py]: Remove `DALI/` here

File: scripts/make_dali_index.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        audio_checksum = md5(
+            os.path.join(data_path, 'audio/', '{}.mp3'.format(trackid)))
+        dali_index[trackid]['audio'].append(audio_checksum)
+        dali_index[trackid]['annot'] = ['DALI/annotations/'+trackid+'.gz']
```
Comment (rabitt) [inline scripts/make_dali_index.py]: Remove `DALI/` here

File: scripts/make_dali_index.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        annot_checksum = md5(
+            os.path.join(data_path, 'annotations/', '{}.gz'.format(trackid)))
+        dali_index[trackid]['annot'].append(annot_checksum)
+
+    with open(DALI_INDEX_PATH, 'w') as fhandle:
+        json.dump(dali_index, fhandle, indent=2)
+
+    return
```
Comment (rabitt) [inline scripts/make_dali_index.py]: you can remove this

File: mirdata/dali.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    """DALI is not available for downloading directly.
+    This function prints a helper message to download DALI
+    through zenodo.org.
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    """
+
+    save_path = utils.get_save_path(data_home)
```
[the remark this answers is not in the document]
Comment (magdalenafuentes) [reply mirdata/dali.py]: I don't know about this one, have to ask Gabriel

**Comment by magdalenafuentes** [comment]
I'm writing the tests and I'm a bit confused with the data types, which type are `words`, `paragraphs` and `annotation_objects`?

File: setup.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 if __name__ == '__main__':
     setup(
         name='mirdata',
-        version='0.0.20',
+        version='0.0.21',
```
Comment (rabitt) [inline setup.py]: change this right before merging, just in case another PR gets merged before this!

**Comment by rabitt** [comment]
> I'm writing the tests and I'm a bit confused with the data types, which type are `words`, `paragraphs` and `annotation_objects`?

@gabolsgabs ?

File: mirdata/dali.py [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+from __future__ import division
+from __future__ import print_function
+
+import json
+import gzip
+import pickle
+import os
+import librosa
+import logging
+import DALI  # this is the package, needed to load the annotations
```
Comment (rabitt) [inline mirdata/dali.py]: let's do `import DALI as dali`

File: setup.py [hunk 7]
```diff
@@ -13,7 +13,7 @@
         long_description="""Common loaders for MIR datasets.""",
         keywords='mir dataset loader audio',
         license='BSD-3-Clause',
-        install_requires=['tqdm', 'librosa >= 0.7.0', 'numpy>=1.16', 'six', 'jams', 'requests'],
+        install_requires=['tqdm', 'librosa >= 0.7.0', 'numpy>=1.16', 'six', 'jams', 'requests', 'dali-dataset'],
```
Comment (rabitt) [inline setup.py]: move this out of install_requires and just keep it in extras_require, but as a separate key, e.g.
```
extras_require = {
   'tests': ...
   'docs': ...
   'dali':  ['dali-dataset']
}
```

And we should add instructions to the docs/readme explaining how to install it - `pip install mirdata[dali]`. We'll have to update the tests to install all the requirements... @drubinstein any thoughts?

File: mirdata/dali.py [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        track_id (str): track id
+        audio_path (str): track audio path
+        artist
```
Comment (rabitt) [inline mirdata/dali.py]: a few missing descriptions

**Comment by lostanlen** [comment]
Hello @magdalenafuentes, there's something i don't understand about this PR. Is the DALI module a subdependency of mirdata or not? From what i see:
* import mirdata.dali fails unless DALI (the package) is installed
* DALI is not used anywhere in the module mirdata.dali
* instead, there is a print statement with the following instruction

```
Use the function dali_code.get_audio you can find at:
        https://github.com/gabolsgabs/DALI for getting the audio and place them at:
```

is this expected?

**Pull request merged**